src/community/generated-geometries/src/main/java/org/geoserver/platform/GeoServerExtensionFinder.java

```diff
@@ -0,0 +1,32 @@
+/* (c) 2019 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+
+package org.geoserver.platform;
+
+import org.springframework.context.ApplicationContext;
+
+import java.io.Serializable;
+import java.util.List;
+
+// TODO: should we move it to "platform" module?
```

Comment:
An interface and implementation were removed from production code. For tests purpose, there are still some lambda-based tools in some constructors but only applied inline in test case classes. There is also a possibility to create objects for tests in a standard way and then production-ready constructors are being used. 
Just for convenience.

To much static stuff is usually a code smell. IMHO we should introduce this kind of decoupling every time it is needed, since its a great help for unit testing and not breaking integration tests.

Reply:
We don't care about theoretical code smells, we care about working software delivered in time that won't take down the server. Different priorities.